registry/purview-registry/script/migrate_legacy_project.py

```diff
@@ -0,0 +1,122 @@
+            print(f"Found existing entity  {entity.guid}, {entity.typeName} -- {entity.qualifiedName}")
+    return UUID(entity.guid)
+default_purview_name = "feathrazuretest3-purview1"   
+default_project_name = "feathr_ci_registry_35_47_50416"                            
+credential = DefaultAzureCredential(exclude_interactive_browser_credential=False)
+oauth = AzCredentialWrapper(credential=credential)
+purview_client = PurviewClient(
+    account_name=default_purview_name,
+    authentication=oauth)
+
+project_search_term = {"entityType": "feathr_workspace_v1"}
+result = purview_client.discovery.query(filter=project_search_term)
+project_entities = [x for x in result['value']]
+
+
+for entity in project_entities:
+    project_entity = purview_client.get_entity(entity["id"])['entities'][0]
+    entities_in_project = []
+    query_filter = {
+        "and": [
+            {
+                "or":
+                [
+                    {"entityType": "feathr_anchor_v1"},
+                    {"entityType": "feathr_source_v1"},
+                    {"entityType": "feathr_anchor_feature_v1"},
+                    {"entityType": "feathr_derived_feature_v1"}
+                ]
+            },
+            {
+                "attributeName": "qualifiedName",
+                "operator": "startswith",
+                "attributeValue": project_entity['attributes']['qualifiedName'] + "__"
+            }
+        ]
+    }
+    entities_in_project = purview_client.discovery.query(filter=query_filter)['value']     
+               
+    anchors = [x for x in entities_in_project if x['entityType']=='feathr_anchor_v1']
+    sources = [x for x in entities_in_project if x['entityType']=='feathr_source_v1']
+    anchor_features = [x for x in entities_in_project if x['entityType']=='feathr_anchor_feature_v1']
+    derived_features = [x for x in entities_in_project if x['entityType']=='feathr_derived_feature_v1']
+
+    for source in sources:
+        source_entity = purview_client.get_entity(source['id'])['entities'][0]
+        relations = _generate_relation_pairs(project_entity,source_entity,Label_Contains)
+        [upload_single_entity_to_purview(purview_client,x) for x in relations]
+    for anchor in anchors:
+        anchor_entity = purview_client.get_entity(anchor['id'])['entities'][0]
+        source_entity = None
+        relations=[]
+        relations+= _generate_relation_pairs(project_entity,anchor_entity,Label_Contains)
+        if anchor_entity['attributes']['source']:
+            source_entity = purview_client.get_entity(anchor_entity['attributes']['source']['guid'])['entities'][0]
+            relations+= _generate_relation_pairs(anchor_entity,source_entity,Label_Consumes)
+
+        containing_anchor_features = [x for x in anchor_features if x['qualifiedName'].startswith(project_entity['attributes']['qualifiedName'] + "__"+anchor['name'])]
```

Comment:
> don't use the hard coded `__` here since it might be changed?

Moved to local var and put it at top of the file

Reply:
> I suggest using the lineage API to get the relationship to avoid those hard coded processing.

lineage api currently does not cover all "contains" relationship, if an orphan entity exist in old format, lineage might not be able to pick it up. (For example, a source that is not consumed by anything in the project?)